pkg/util/kubernetes/autoscalers/processor.go

```diff
@@ -146,13 +151,25 @@ func (p *Processor) ProcessWPAs(wpa *v1alpha1.WatermarkPodAutoscaler) map[string
 }
 
 func makeChunks(batch []string) (chunks [][]string) {
-	for i := 0; i < len(batch); i += chunkSize {
-		if i+chunkSize > len(batch) {
-			chunks = append(chunks, batch[i:])
-			break
+	// uriLength is used to avoid making a query that goes beyond the maximum URI size.
+	var uriLength int
+	var tempBucket []string
+
+	for _, val := range batch {
+		// Length of the query plus comma, time and space aggregators that come later on.
+		tempSize := len(url.PathEscape(val)) + extraQueryCharacters
+		uriLength = uriLength + tempSize
+		if uriLength >= maxCharactersPerChunk || len(tempBucket) >= chunkSize {
```

Comment:
where do you subtract the URI prefix (`https://api.datadoghq.com/api/`) size?

Reply:
It is accounted for in the `maxCharactersPerChunk`. Because I have no control over the other parameters that are added in the url (base, from-to, api key...), I took a conservative value (from my testings, we hit the issue when the query is > `7607` escaped characters).

Reply:
ok great, I didn't know that the official limit is: 7607

we should put the information in cluster-agent document. If a user is also using a proxy, the URI limit in the default proxy configuration can be lower.